src/ObjectUtils/CoffFile.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ObjectUtils/CoffFile.h"
+
+#include <absl/strings/str_format.h>
+#include <llvm/Demangle/Demangle.h>
+#include <llvm/Object/Binary.h>
+#include <llvm/Object/COFF.h>
+#include <llvm/Object/ObjectFile.h>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Result.h"
+#include "symbol.pb.h"
+
+namespace orbit_object_utils {
+
+namespace {
+
+using orbit_grpc_protos::ModuleSymbols;
+using orbit_grpc_protos::SymbolInfo;
+
+class CoffFileImpl : public CoffFile {
+ public:
+  CoffFileImpl(std::filesystem::path file_path,
+               llvm::object::OwningBinary<llvm::object::ObjectFile>&& owning_binary);
+  [[nodiscard]] ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> LoadDebugSymbols() override;
+  [[nodiscard]] bool HasDebugSymbols() const override;
+  [[nodiscard]] std::string GetName() const override;
+  [[nodiscard]] std::filesystem::path GetFilePath() const override;
```

Comment:
should this be a const-ref?

Reply:
I think it could be. That would mean this has to be purely a getter of the member, right? This is one of the methods that was already there and I took the existing signature, so not sure if there was a deeper reason why this is not const&. Also, our codebase seems to not do this consistently throughout.

Reply:
I am not sure why it is returned by value. Probably because in this case this is an interface and we cannot speculate about implementation. If it stores the value or generates it based on something else. My personal preference here would be to use const-ref until proven otherwise.

Reply:
Done.